### PR TITLE
add batch/recent endpoint

### PR DESCRIPTION
**Pull request by cseed**
informational page for last 100 job created (with command line) and job finished (with log) events

This is for @jigold's demo on Monday so she can give some sense of progress of jobs when running a pipeline demo.

Note, batch isn't exposed publicly so you'll have to `kubectl` port forward to access it.

Not the most beautiful, but should be functional.  Here's a screenshot:

screenshot

### CODE REVIEW

File: batch/batch/server/globals.py [hunk 1]
```diff
@@ -24,3 +26,19 @@ def next_id():
...
+def get_recent_events():
+    return _recent_events
+
+
+def add_event(event):
+    event['time'] = str(datetime.datetime.now())
+
+    global _recent_events
+
+    _recent_events = _recent_events[:100]
```
Comment (tpoterba) [inline batch/batch/server/globals.py]: should be a deque: https://docs.python.org/3.3/library/collections.html#collections.deque

File: batch/batch/server/server.py [hunk 2]
```diff
@@ -142,6 +142,8 @@ def __init__(self, pod_spec, batch_id, attributes, callback, parent_ids):
             spec=pod_spec)
 
         log.info('created job {}'.format(self.id))
+        print('pod_spec', pod_spec.containers[0].command)
```
Comment (tpoterba) [inline batch/batch/server/server.py]: stray print

File: batch/batch/server/server.py [hunk 3]
```diff
@@ -533,6 +538,13 @@ def refresh_k8s_state():
     return '', 204
 
 
+@app.route('/recent', methods=['GET'])
+def recent():
+    recent_events = get_recent_events()
+    print(recent_events)
```
Comment (tpoterba) [inline batch/batch/server/server.py]: stray print

**Comment by cseed** [comment]
Thanks, good suggestion.  I also backed off the number of recent events at 50 and capped the pod_log at 64K.

**Pull request merged**